bin/yarn.js

```diff
@@ -4,41 +4,31 @@
 /* eslint-disable flowtype/require-valid-file-annotation */
 'use strict';
 
-// validate that used node version is supported
-var semver = require('semver');
 var ver = process.versions.node;
-ver = ver.split('-')[0]; // explode and truncate tag from version #511
+var majorVer = parseInt(ver.split('.')[0], 10);
 
 var dirPath = null;
 
-if (semver.satisfies(ver, '>=5.0.0')) {
-  dirPath = '../lib';
-} else if (semver.satisfies(ver, '>=4.0.0')) {
-  dirPath = '../lib-legacy';
+if (majorVer >= 5) {
+  dirPath = '../lib/';
+} else if (majorVer >= 4) {
+  dirPath = '../lib-legacy/';
 } else {
-  console.log(require('chalk').red('Node version ' + ver + ' is not supported, please use Node.js 4.0 or higher.'));
-  // We need to skip the rest and terminate here since the Node version is not supported.
-  // `console.log` calls are sync so we don't have a risk of exiting early before stdout flushes
-  process.exit(1); // eslint-disable-line no-process-exit
+  console.error('Node version ' + ver + ' is not supported, please use Node.js 4.0 or higher.');
+  process.exitCode = 1;
 }
 
-// load v8-compile-cache
-if (semver.satisfies(ver, '>=5.7.0')) {
-  require('v8-compile-cache');
+if (dirPath) {
+  var v8CompileCachePath = dirPath + 'v8-compile-cache';
+  var fs = require('fs');
+  // We don't have/need this on legacy builds and dev builds
+  fs.existsSync(v8CompileCachePath) && require(dirPath + 'v8-compile-cache');
+
+  // Just requiring this package will trigger a yarn run since the
+  // `require.main === module` check inside `cli/index.js` will always
+  // be truthy when built with webpack :(
+  var cli = require(dirPath + 'cli');
+  if (!cli.webpackPolyfill) {
```

Comment:
I think it can be. I'm not super happy with the approach but I didn't want to add more code/variables etc. either. We can improve this in a follow-up.